### DISCUSSION TITLE
Apple Silicon?

### DISCUSSION THREAD

**Discussion by benjamind2008** [revvv/gdash-export-CrLi#90] [Ideas]
I do believe you can run it under Rosetta if it's the x86 Mac version, but are there any plans by anyone to fork GDash and code an Apple Silicon version? I'm unsure what graphics engine they'd use, but I was to believe Metal is the graphics pipeline they use for silicon Mx GPUs as they have direct access to most levels of abstraction.

It would be nice to see someone code an Apple silicon version of GDash. Meanwhile I love the Windows version, it runs so smoothly and without any problems. I would keep my PC just to run this masterpiece!

**Comment by revvv** [comment]
Thank you for your kind words. Right now I can't compile it for Apple Silicon, because I don't have the hardware.
AFAIK it is possible to run the x86 version on newer Macs. I guess, you would  not notice a difference if it's x86 or Mx.

**Comment by benjamind2008** [comment]
Thanks for the clarification. Would the x86 Mac version use OpenGL by any
chance? From what I gather OpenGL is still being used on OSX.

**Comment by revvv** [comment]
The x86 Mac version uses OpenGL if GDash is configured accordingly, but you probably wanted to know if the x86 version on Mx can use OpenGL? I don't know, because I have no new Mac. @cwscws can you answer this?

**Comment by cwscws** [comment]
The Mac version works perfectly on my iMac M1 and also on my Mac Mini M4 Pro. OpenGL also works perfectly, even with several of the GL shaders.

I would also like to see a native Apple Silicon built in future, you can never know if or when Apple drops x86 emulation as they have done it before…

Native Metal graphics API would be very much appreciated as Apple does not support OpenGL anymore, it’s only a matter of time when it’s not in one of the next OS updates anymore. Best would be BGFX (like MAME or Visual Pinball lately).
